packages/airnode-utilities/src/gas-prices/gas-prices.ts

```diff
@@ -1,15 +1,21 @@
 import { BigNumber, ethers } from 'ethers';
-import { go, logger, PendingLog } from '@api3/airnode-utilities';
-import { BASE_FEE_MULTIPLIER, DEFAULT_RETRY_TIMEOUT_MS, PRIORITY_FEE } from '../constants';
-import { GasTarget, LogsData } from '../types';
-import { ChainOptions, PriorityFee } from '../config/types';
+import { ChainOptions, PriorityFee, GasTarget } from './types';
+import { go, logger, PendingLog } from '../';
+import { LogsData } from '../logging';
+
+// The default amount of time before a "retryable" promise is timed out and retried
+export const DEFAULT_RETRY_TIMEOUT_MS = 5_000;
+// The Priority Fee in Wei
+export const PRIORITY_FEE = 3120000000;
+// The Base Fee to Max Fee multiplier
+export const BASE_FEE_MULTIPLIER = 2;
 
 export interface FetchOptions {
   readonly provider: ethers.providers.JsonRpcProvider | ethers.providers.Provider;
   readonly chainOptions: ChainOptions;
 }
 
-export const parsePriorityFee = ({ value, unit }: PriorityFee) =>
+export const parsePriorityFee = ({ value, unit }: PriorityFee): BigNumber =>
   ethers.utils.parseUnits(value.toString(), unit ?? 'wei');
 
 export const getLegacyGasPrice = async (options: FetchOptions): Promise<LogsData<GasTarget | null>> => {
```

Comment:
We should probably make `DEFAULT_RETRY_TIMEOUT_MS` configurable with an optional argument. An optional argument won't break existing usages. WDYT? Everything else is exposed (the default constants for gas pricing are just defaults/they're configurable through function arguments).

Reply:
Great idea, would it make sense to add the whole `go` [PromiseOptions](https://github.com/api3dao/airnode/blob/master/packages/airnode-utilities/src/promises/promise.ts#L12) as a field to `FetchOptions`
```js
export interface FetchOptions {
  readonly provider: ethers.providers.JsonRpcProvider | ethers.providers.Provider;
  readonly chainOptions: ChainOptions;
  readonly promiseOptions?: PromiseOptions;
}
```

and then for example:
```js
export const getLegacyGasPrice = async (options: FetchOptions): Promise<LogsData<GasTarget | null>> => {
  const { provider, promiseOptions } = options;

  const [err, gasPrice] = await go(() => provider.getGasPrice(), {
    retries: 1,
    timeoutMs: DEFAULT_RETRY_TIMEOUT_MS,
    ...promiseOptions,
  });
  if (err || !gasPrice) {
    const log = logger.pend('ERROR', 'All attempts to get legacy gasPrice from provider failed');
    return [[log], null];
  }

  return [[], { gasPrice }];
};
```

Reply:
`FetchOptions` should probably be moved to `./types` as well :smile:

Reply:
Maybe it's worth merging `chainOptions` and `promiseOptions` into a single object?

Reply:
@andreogle Did you mean as a field in chainOptions :detective: ?
```js
export interface ChainOptions {
  txType: 'legacy' | 'eip1559';
  baseFeeMultiplier?: number;
  priorityFee?: PriorityFee;
  promiseOptions?: PromiseOptions;
}
```
and then:
```js
const [err, gasPrice] = await go(() => provider.getGasPrice(), {
    retries: 1,
    timeoutMs: DEFAULT_RETRY_TIMEOUT_MS,
    ...chainOptions.promiseOptions,
  });
```

Reply:
I don't see why not :smile: